src/macros/macro.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <QtCore>
+
+#include "proto/macro.pb.h"
+namespace proto = mixxx::track::io;
+
+const QLoggingCategory macroLoggingCategory("macros");
+
+/// A MacroAction is the building block of a Macro.
+/// It contains a position as well as the action to be taken at that position.
+///
+/// Note that currently only jumps to a target position are available, but that
+/// is subject to change.
+struct MacroAction {
+    MacroAction(){};
+    MacroAction(double position, double target)
+            : position(position), target(target){};
+    /// TODO(xerus) use FramePos once https://github.com/mixxxdj/mixxx/pull/2861 is merged
+    double position;
+    double target;
+
+    bool operator==(const MacroAction& other) const {
+        return position == other.position && target == other.target;
+    }
+
+    inline bool operator!=(const MacroAction& other) const {
+        return !operator==(other);
+    }
+
+    proto::Macro_Action* serialize() const {
+        auto serialized = new proto::Macro_Action();
+        serialized->set_origin(position);
+        serialized->set_target(target);
+        serialized->set_type(Type::JUMP);
+        return serialized;
+    };
+
+    enum Type : uint8_t {
+        JUMP = 0
+    };
+};
+
+/// A Macro stores a list of MacroActions.
+/// The maximum size is fixed at a generous kMaxSize to prevent resizing in realtime code.
+class Macro {
+  public:
+    static const int kMaxSize = 1000;
+
+    Macro() = default;
+    Macro(QByteArray serialized);
+
+    MacroAction actions[kMaxSize];
```

Comment:
`m_actions` and this should by private IMHO. If you use a SPSC queue in the macro recorder, you can just move the recording logic out of the macro class itself and let the macro recorder create the Macro instance once recording is finished. Then this can become QList or something like that with a getter that contains a const reference.